blueprints/cloud-operations/network-dashboard/cloud-function/metrics/secondarys.py

```diff
@@ -0,0 +1,273 @@
+      pod_full_path = f"{pod_parent}/{host_node_name}"
+
+      # A check to make sure pod is not using node IP
+      if cur_PodIP != cur_HostIP:
+        node_secondary_dict[pod_full_path]['used_IP'] += 1
+
+  # Counting IP addresses used by Service in GKE
+  response_service = config["clients"]["asset_client"].search_all_resources(
+      request={
+          "scope": f"organizations/{config['organization']}",
+          "asset_types": ['k8s.io/Service'],
+          "read_mask": read_mask,
+          "page_size": config["page_size"],
+      })
+
+  for asset in response_service:
+    service_parent = "/".join(asset.name.split('/')[4:9])
+    service_fullpath = f"{service_parent}/Service"
+    cluster_secondary_dict[service_fullpath]['used_ip_addresses'] += 1
+
+  for item in node_secondary_dict.values():
+    itemKey = f"{item['node_parent']}/{item['this_node_pool']}"
+    cluster_secondary_dict[itemKey]['used_ip_addresses'] += item['used_IP']
+
+  for item in cluster_secondary_dict.values():
+    itemKey = f"{item['subnet']}/{item['secondaryRange_name']}"
+    all_secondary_dict[item['project']][itemKey]['used_ip_addresses'] += item[
+        'used_ip_addresses']
+
+
+def compute_secondary_utilization(config, all_secondary_dict):
+  '''
+    Counts resources (GKE, GCE) using IPs in secondary ranges.
+      Parameters:
+        config (dict): Dict containing config like clients and limits
+        all_secondary_dict (dict): Dict containing the secondary IP Range information for each subnets in the GCP organization
+      Returns:
+        None
+  '''
+  read_mask = field_mask_pb2.FieldMask()
+  read_mask.FromJsonString('name,versionedResources')
+
+  compute_GKE_secondaryIP_utilization(config, read_mask, all_secondary_dict)
+  # TODO: Other Secondary IP like GCE VM using alias IPs
+
+
+def get_secondaries(config, metrics_dict):
+  '''
+    Writes all secondary rang IP address usage metrics to custom metrics.
+      Parameters:
+        config (dict): The dict containing config like clients and limits
+      Returns:
+        None
+  '''
+
+  secondaryRange_dict = get_all_secondaryRange(config)
+  # Updates all_subnets_dict with the IP utilization info
+  compute_secondary_utilization(config, secondaryRange_dict)
+
+  timestamp = time.time()
+  for project_id in config["monitored_projects"]:
+    if project_id not in secondaryRange_dict:
+      continue
+    for secondary_dict in secondaryRange_dict[project_id].values():
+      ip_utilization = 0
+      if secondary_dict['used_ip_addresses'] > 0:
+        ip_utilization = secondary_dict['used_ip_addresses'] / secondary_dict[
+            'total_ip_addresses']
+
+      # Building unique identifier with subnet region/name
+      subnet_id = f"{secondary_dict['region']}/{secondary_dict['name']}"
+      metric_labels = {
+          'project': project_id,
+          'network_name': secondary_dict['network_name'],
+          'subnet_id': subnet_id
```

Comment:
maybe add one label with the region + subnet name and another label with the region + secondary range name?